cloud/scope/machine.go

```diff
@@ -477,45 +488,96 @@ func (m *MachineScope) ReconcileCreateInstanceOnLB(ctx context.Context) error {
+					LoadBalancerId: loadbalancerId,
+					BackendSetName: backendSet.Name,
+					CreateBackendDetails: loadbalancer.CreateBackendDetails{
+						IpAddress: common.String(instanceIp),
+						Port:      common.Int(int(m.OCICluster.Spec.ControlPlaneEndpoint.Port)),
+					},
+					OpcRetryToken: ociutil.GetOPCRetryToken("%s-%s", "create-backend", string(m.OCIMachine.UID)),
+				})
+				if err != nil {
+					return err
+				}
+				m.OCIMachine.Status.CreateBackendWorkRequestId = *resp.OpcWorkRequestId
+				logger.Info("Add instance to LB backend-set", "WorkRequestId", resp.OpcWorkRequestId)
+				logger.Info("Waiting for LB work request to be complete")
+				_, err = ociutil.AwaitLBWorkRequest(ctx, m.LoadBalancerClient, resp.OpcWorkRequestId)
+				if err != nil {
+					return err
+				}
 			}
-			m.OCIMachine.Status.CreateBackendWorkRequestId = *resp.OpcWorkRequestId
-			logger.Info("Add instance to LB backend-set", "WorkRequestId", resp.OpcWorkRequestId)
-			logger.Info("Waiting for LB work request to be complete")
-			_, err = ociutil.AwaitLBWorkRequest(ctx, m.NetworkLoadBalancerClient, resp.OpcWorkRequestId)
-			if err != nil {
-				return err
+		}
+
+	} else {
+		lb, err := m.NetworkLoadBalancerClient.GetNetworkLoadBalancer(ctx, networkloadbalancer.GetNetworkLoadBalancerRequest{
+			NetworkLoadBalancerId: loadbalancerId,
+		})
+		if err != nil {
+			return err
+		}
+		backendSet := lb.BackendSets[APIServerLBBackendSetName]
+		if !m.containsNLBBackend(backendSet, m.Name()) {
+			logger := m.Logger.WithValues("backend-set", *backendSet.Name)
+			logger.Info("Checking work request status for create backend")
+			workRequest := m.OCIMachine.Status.CreateBackendWorkRequestId
+			if workRequest != "" {
+				_, err = ociutil.AwaitNLBWorkRequest(ctx, m.NetworkLoadBalancerClient, &workRequest)
+				if err != nil {
+					return err
+				}
+			} else {
+				resp, err := m.NetworkLoadBalancerClient.CreateBackend(ctx, networkloadbalancer.CreateBackendRequest{
+					NetworkLoadBalancerId: loadbalancerId,
+					BackendSetName:        backendSet.Name,
+					CreateBackendDetails: networkloadbalancer.CreateBackendDetails{
+						IpAddress: common.String(instanceIp),
+						Port:      common.Int(int(m.OCICluster.Spec.ControlPlaneEndpoint.Port)),
+						Name:      common.String(m.Name()),
+					},
+					OpcRetryToken: ociutil.GetOPCRetryToken("%s-%s", "create-backend", string(m.OCIMachine.UID)),
+				})
+				if err != nil {
+					return err
+				}
+				m.OCIMachine.Status.CreateBackendWorkRequestId = *resp.OpcWorkRequestId
+				logger.Info("Add instance to NLB backend-set", "WorkRequestId", resp.OpcWorkRequestId)
+				logger.Info("Waiting for NLB work request to be complete")
+				_, err = ociutil.AwaitNLBWorkRequest(ctx, m.NetworkLoadBalancerClient, resp.OpcWorkRequestId)
+				if err != nil {
+					return err
+				}
+				logger.Info("NLB Backend addition work request is complete")
```

Comment:
More of a note for later, but I think we can clean up this function and possibly DRY up some code here.